server/controllers/intreview/cancalIntreview.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable space-before-blocks */
+import { Response } from 'express';
+import { RequestType, mailSender } from '../../utils';
+import Interviewee from '../../database/Models/Interviewee';
+import Interviewer from '../../database/Models/Interviewer';
+import User from '../../database/Models/User';
+
+const cancalInterview = async (req: RequestType, res: Response) => {
+  // use vaildation for params after marge
+  const { interviewId } : any = req.params;
+  const { userInfo } = req;
+
+  const user = await User.findOne({ _id: userInfo?.id }, { email: 1, _id: 1, name: 1 });
+  const { email } = user;
+
+  // eslint-disable-next-line max-len
+  let collectionName: any; // the name of the collection (intrviewee or interviewer)
+
+  // // query about role by _id
+  if (userInfo){
+    if (userInfo.role === 'interviewee') {
+      collectionName = Interviewee;
+    } else if (userInfo.role === 'interviewer') {
+      collectionName = Interviewer;
+    }
+  }
+  const result = await collectionName.updateOne({ 'interviews._id': interviewId }, { $set: { 'interviews.$.is_cancalled': true } });
```

Comment:
You will cancel the interview in both interviewee and interview so no need for the above conditions.